modules/perception/lidar/lib/roi_filter/hdmap_roi_filter/polygon_scan_cvter.h

```diff
@@ -316,23 +312,23 @@ void PolygonScanCvter<T>::ParsePolygon(const DirectionMajor dir_major,
 
     // get segment
     Segment line_seg(vertex, nex_vertex);
-    double x_diff = (nex_x - x);
-    double y_diff = (nex_y - y);
+    double x_diff = nex_x - x;
+    double y_diff = nex_y - y;
 
     // get k
     segments.push_back(line_seg);
-    std::abs(x - nex_x) < s_epsilon_ ? ks.push_back(s_inf_)
-                                     : ks.push_back(y_diff / x_diff);
-    double pre_x_diff = (pre_x - x);
+    std::abs(x_diff) < s_epsilon_ ? ks.push_back(s_inf_)
+                                  : ks.push_back(y_diff / x_diff);
+    double pre_x_diff = x - pre_x;
 
     // get singular property
     // ensure fill edge
     // case for zero
-    if (std::abs(x - nex_x) < s_epsilon_ || std::abs(pre_x_diff) < s_epsilon_) {
+    if (std::abs(x_diff) < s_epsilon_ || std::abs(pre_x_diff) < s_epsilon_) {
       is_singular.push_back(true);
     } else {
-      pre_x_diff* x_diff > 0 ? is_singular.push_back(true)
-                             : is_singular.push_back(false);
+      pre_x_diff * x_diff < 0 ? is_singular.push_back(true)
```

Comment:
Doesn’t need to be changed

Reply:
Just want to make the logic clearer. Indeed, it doesn't matter. Will change it back.